### DISCUSSION TITLE
Is there anyway to check the current syntax on  cursor is error

### DISCUSSION THREAD

**Discussion by windwp** [nvim-treesitter/nvim-treesitter#1274] [Q&A]
Sample 
``` lua
function abcde(target)
    if data then<cursor>
      ............
end
```
VS
``` lua
function abcde(target)
    if data then<cursor>
     .................
   end
end
```
So on cursor the syntax is error but is there any way to know that with out lsp and use tree sitter. but when i try to get node on the  cursor
treesitter always group the error node with the node below so it is too difficult to know error.

https://github.com/windwp/nvim-autopairs/blob/master/doc/endwise.md

**Comment by vigoux** [comment]
Tree-sitter is doing this, grouping the error in one big error node.

We can't customize that though, so we can't do anything actually here.
